### PR TITLE
Remove spinning for blocked socket ops

### PR DESCRIPTION
**Pull request by benaadams**
As per https://github.com/dotnet/corefx/pull/5655

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class SocketInput : ICriticalNotifyCompletion
 
         private readonly MemoryPool2 _memory;
         private readonly IThreadPool _threadPool;
-        private readonly ManualResetEventSlim _manualResetEvent = new ManualResetEventSlim(false);
+        private readonly ManualResetEventSlim _manualResetEvent = new ManualResetEventSlim(false, 0);
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs]: In CoreFX case, they seem to have removed `false` also... should it be the same here?

Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs]: They added `false` :)

Comment (muratg) [inline src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs]: Yup, my bad :smile:

**Comment by cesarbs** [comment]
LGTM

**Pull request merged**